src/lib/teads-curve/index.ts

```diff
@@ -0,0 +1,134 @@
+import {IImpactModelInterface} from '../interfaces';
+import Spline from 'typescript-cubic-spline';
+import {KeyValuePair} from '../../types/boavizta';
+
+export class TeadsCurveModel implements IImpactModelInterface {
+  // Defined for compatibility. Not used in TEADS.
+  authParams: object | undefined;
+  // name of the data source
+  name: string | undefined;
+  // tdp of the chip being measured
+  tdp = 100;
+  // default power curve provided by the Teads Team
+  curve: number[] = [0.12, 0.32, 0.75, 1.02];
```

Comment:
let's keep such configurations in config file

Reply:
it's convenient when f.ex. you want to tune boavizta params, you go directly to config and you know that everything should be tuned from there instead of diving into codebase.

Reply:
These plugins will be spinoff into a separate project sooner.  

The main point is that this configuration is adjustable while using the Impact Plugin. 

It will be unnecessary to extract them to a separate file, as these are **not constants** and are modified while using the plugin. 

These should be documented into the documentation rather than being extracted to a separate file for readability.

Reply:
Got your point. So these values are dynamic rather than static.

Reply:
In this case these are a magic set of numbers that are called the TEADs curve. So they won't ever change. Happy for them to remain as is.

Reply:
@jawache I intend to allow anyone to adjust the spline. Just keeping an option open at one end so plugin users can adjust instead of creating their own version of the implementation.